kornia/augmentation/augmentation.py

```diff
@@ -1723,3 +1724,80 @@ def apply_transform(
         self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
         return input + params['noise'].to(input.device) * self.std + self.mean
+
+
+class RandomFisheye(AugmentationBase2D):
+    r"""Add random camera radial distortion.
+
+    Args:
+        center_x (torch.Tensor): Ranges to sample respect to x-coordinate center with shape (2,).
+        center_y (torch.Tensor): Ranges to sample respect to y-coordinate center with shape (2,).
+        gamma (torch.Tensor): Ranges to sample for the gamma values respect to optical center with shape (2,).
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+            input tensor. If ``False`` and the input is a tuple the applied transformation wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        p (float): probability of applying the transformation. Default value is 0.5.
+
+    Examples:
+        >>> img = torch.ones(1, 1, 2, 2)
+        >>> center_x = torch.tensor([-.3, .3])
+        >>> center_y = torch.tensor([-.3, .3])
+        >>> gamma = torch.tensor([-1., 1.])
+        >>> out = RandomFisheye(center_x, center_y, gamma)(img)
+        >>> out.shape
+        torch.Size([1, 1, 2, 2])
+    """
+
+    def __init__(self,
+                 center_x: torch.Tensor,
+                 center_y: torch.Tensor,
+                 gamma: torch.Tensor,
+                 return_transform: bool = False,
+                 same_on_batch: bool = False,
+                 p: float = 0.5) -> None:
+        super(RandomFisheye, self).__init__(
+            p=p, return_transform=return_transform, same_on_batch=same_on_batch, p_batch=1.)
+        self.center_x = self._check_tensor(center_x)
+        self.center_y = self._check_tensor(center_y)
+        self.gamma = self._check_tensor(gamma)
+        self.dist = torch.distributions.Uniform
+
+    def __repr__(self) -> str:
+        return self.__class__.__name__ + f"({super().__repr__()})"
+
+    def _check_tensor(self, data: torch.Tensor) -> torch.Tensor:
+        if not isinstance(data, torch.Tensor):
+            raise TypeError(f"Invalid input type. Expected torch.Tensor - got: {type(data)}")
+
+        if len(data.shape) != 1 and data.shape[0] != 2:
+            raise ValueError(f"Tensor must be of shape (2,). Got: {data.shape}.")
+
+        return data
+
+    def generate_parameters(self, shape: torch.Size) -> Dict[str, torch.Tensor]:
+        B, _, _, _ = shape  # batch_size
+        center_x = self.dist(self.center_x[:1], self.center_x[1:]).rsample(shape[:1])
+        center_y = self.dist(self.center_y[:1], self.center_y[1:]).rsample(shape[:1])
```

Comment:
We should pre-build the distribution modules and only call resample in this function, to avoid creating distribution modules on the fly.

Reply:
Ok, we can do it with the new API

Reply:
We might init two uniform distributions from 0-1 in the initializer. Then scale it to the proper range in the param generator.